src/installer/managed/Microsoft.NET.HostModel/Bundle/Manifest.cs

```diff
@@ -107,6 +120,28 @@ public FileEntry AddEntry(FileType type, string relativePath, long offset, long
             return entry;
         }
 
+        private static byte[] ComputeSha256Hash(Stream stream)
+        {
+            using (SHA256 sha = SHA256.Create())
+            {
+                return sha.ComputeHash(stream);
+            }
+        }
+
+        private string GetDeterministicId()
```

Comment:
Given that the BundleId is accessed only from Write (and then tests) I think this should be called directly from Write instead. There's no need to support calling this before Write is called. You can then also remove the need for another field.

It would make sense externally - call AddEntry* and once done, call Write. And that's when everything is set in stone (and you get an ID as well).

Reply:
Makes sense.
Caller of this probably should know well what needs to be called.
I'm cleaning this...